src/app/common/services/view-util.service.ts

```diff
@@ -0,0 +1,194 @@
+import {Injectable} from '@angular/core';
+import {AlfrescoApiService, LogService} from '@alfresco/adf-core';
+import {MinimalNodeEntity, RenditionEntry} from 'alfresco-js-api';
+import {ContentApiService} from '../../services/content-api.service';
+
+@Injectable()
+export class ViewUtilService {
+    static TARGET = '_new';
+
+    /**
+     * Content groups based on categorization of files that can be viewed in the web browser. This
+     * implementation or grouping is tied to the definition the ng component: ViewerComponent
+     * @type {{IMAGE: string; MEDIA: string; PDF: string; TEXT: string}}
+     */
+    public static ContentGroup = {
+        IMAGE: 'image',
+        MEDIA: 'media',
+        PDF: 'pdf',
+        TEXT: 'text'
+    };
+
+    /**
+     * Based on ViewerComponent Implementation, this value is used to determne how many times we try
+     * to get the rendition of a file for preview, or printing.
+     * @type {number}
+     */
+    maxRetries = 5;
+
+    /**
+     * Mimetype grouping based on the ViewerComponent.
+     * @type {{text: string[]; pdf: string[]; image: string[]; media: string[]}}
+     */
+    private mimeTypes = {
+        text: ['text/plain', 'text/csv', 'text/xml', 'text/html', 'application/x-javascript'],
+        pdf: ['application/pdf'],
+        image: ['image/png', 'image/jpeg', 'image/gif', 'image/bmp', 'image/svg+xml'],
+        media: ['video/mp4', 'video/webm', 'video/ogg', 'audio/mpeg', 'audio/ogg', 'audio/wav']
+    };
+
+    constructor(private apiService: AlfrescoApiService,
+                private contentApi: ContentApiService,
+                private logService: LogService) {
+    }
+
+    /**
+     * This method takes a url to trigger the print dialog against, and the type of artifact that it
+     * is.
+     * This URL should be one that can be rendered in the browser, for example PDF, Image, or Text
+     * @param {string} url
+     * @param {string} type
+     */
+    public printFile(url: string, type: string) {
+        const pwa = window.open(url, ViewUtilService.TARGET);
+        // Because of the way chrome focus and close image window vs. pdf preview window
+        if (type === ViewUtilService.ContentGroup.IMAGE) {
+            pwa.onfocus = function () {
```

Comment:
can we use arrow function and extract ```onfocus``` to a single method since its the same as [here](https://github.com/Alfresco/alfresco-content-app/pull/518/files#diff-8e22dbbfdcd5a9a288da487bbc6f0eefR68)